src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.AsyncProjectWorkItemQueue.cs

```diff
@@ -75,7 +75,7 @@ protected override bool AddOrReplace_NoLock(WorkItem item)
                     if (_projectWorkQueue.TryGetValue(key, out var existingWorkItem))
                     {
                         // replace it.
-                        _projectWorkQueue[key] = existingWorkItem.With(item.InvocationReasons, item.ActiveMember, item.Analyzers, item.IsRetry, item.AsyncToken);
+                        _projectWorkQueue[key] = existingWorkItem.With(item.InvocationReasons, item.ActiveMember, item.SpecificAnalyzers, item.IsRetry, item.AsyncToken);
```

Comment:
what's the diff between Analyzers and SpecificAnalyzers?

Reply:
So, I was confused into thinking that `Analyzers` field on a work item always has a list of one or more analyzers that are applicable for the work item. On further looking at the code, it seems this field is non-zero only if the work item was created for a specific analyzer. If it is empty, it indicates that the work item is not for specific analyzer(s), and should execute for all analyzers belonging to the work co-ordinator queue executing it. I felt it was more intuitive to rename the field, and I have also updated the doc comments on it. Additionally, I have also added an API `WorkItem.GetApplicableAnalyzers(allAnalyzers)` that should be used by each queue to find analyzers to execute for any given work item.